diff -up fsplib-0.14/SConstruct.orig fsplib-0.14/SConstruct
--- fsplib-0.14/SConstruct.orig	2019-08-16 08:29:00.000000000 +0200
+++ fsplib-0.14/SConstruct	2023-06-04 13:37:10.122039427 +0200
@@ -11,8 +11,6 @@ PREFIX='/usr/local'
 SHARED=0
 
 env = Environment()
-# Turn CPPFLAGS to list
-env.Append( CPPFLAGS = [])
 
 ############## Imports #################
 from maintainer import checkForMaintainerMode
@@ -40,6 +38,20 @@ def importEnv(list=None, prefix=None):
                kw={ 'ENV': kw }
                env.Append(**kw)
 
+importEnv(['CPPFLAGS','CCFLAGS'])
+if env['ENV'].get('CPPFLAGS'):
+    env.Append( CPPFLAGS = env['ENV'].get('CPPFLAGS').split() )
+elif ARGUMENTS.get('CPPFLAGS', 0):
+    env.Append( CPPFLAGS = ARGUMENTS.get('CPPFLAGS').split() )
+else:
+    env.Append( CPPFLAGS = [])
+if env['ENV'].get('CFLAGS'):
+    env.Append( CCFLAGS = env['ENV'].get('CFLAGS').split() )
+elif ARGUMENTS.get('CFLAGS', 0):
+    env.Append( CCFLAGS = ARGUMENTS.get('CFLAGS').split() )
+else:
+    env.Append( CCFLAGS = [])
+
 #import environment
 importEnv(['HOME','CC'])
 importEnv(prefix='DISTCC_')
